website/docs/reference/warehouse-profiles/spark-profile.md

```diff
@@ -21,6 +21,11 @@ dbt-spark can connect to Spark clusters by three different methods:
 - `odbc` is the preferred method when connecting to Databricks. It supports connecting to a SQL Endpoint or an all-purpose interactive cluster.
 - `http` is a more generic method for connecting to a managed service that provides an HTTP endpoint. Currently, this includes connections to a Databricks interactive cluster.
 - `thrift` connects directly to the lead node of a cluster, either locally hosted / on premise or in the cloud (e.g. Amazon EMR).
+- `session` *DO NOT USE FOR PRODUCTION WORKLOADS* use a pySpark session.
+
+:::info session
+The `session` connection method is not intended for production workloads because data models might be read into memory.
```

Comment:
@jtcohen6 : The only time this happens when it's not _really_ required is when `fetchone` is called. (Then all data is fetched into memory and the first row is popped from that list.) 

A question would be: why do we consider this connection method to be not ready for production workloads?

Reply:
ok, thanks for your reply, on the points:

- Spark sessions can be configured to connect to a non-local Spark cluster. It does not have to be on the same compute as dbt. 
- I also do not see this being supported in the dbt cloud
- I like the "for advanced users". 

I have updated the PR accordingly